### ISSUE TITLE
IllegalArgumentException: Stacks have different current sizes [0] and [1] when building with ProGuard optimizations

### ISSUE THREAD

**Issue by cypressious** [Kotlin/anko#235]
Pulling this out of #206:

When building with Proguard optimizations on (`getDefaultProguardFile('proguard-android-optimize.txt')`), I get the following error:

```
java.lang.IllegalArgumentException: Stacks have different current sizes [0] and [1]
Unexpected error while performing partial evaluation:
Class = [org/jetbrains/anko/AsyncKt$doAsync$1]
Method = [invoke()Ljava/lang/Object;]
Exception = java.lang.IllegalArgumentException
```

This is also reported in https://youtrack.jetbrains.com/issue/KT-12866 and https://sourceforge.net/p/proguard/bugs/610/.

The problem lies in the usage of the elvis operator in the catch block in https://github.com/Kotlin/anko/blob/f1f3033d18f978774d8b6a5a09c73e151d134f11/anko/library/static/common/src/Async.kt#L140

@yanex Although this is technically a ProGuard bug, would you consider rewriting the code without the elvis operator and releasing a bugfix build? In the meantime, we're not able to use ProGuard optimizations in our app.

**Comment by yanex** [comment]
Will be fixed in 0.10.

**Comment by dlazerka** [comment]
Not sure it it's the same problem, or a bug in Proguard:
```
 [proguard] Unexpected error while performing partial evaluation:
 [proguard]   Class       = [kotlin/reflect/jvm/internal/impl/renderer/DescriptorRendererImpl]
 [proguard]   Method      = [renderPossiblyInnerType(Ljava/lang/StringBuilder;Lkotlin/reflect/jvm/internal/impl/descriptors/PossiblyInnerType;)V]
 [proguard]   Exception   = [java.lang.IllegalArgumentException] (Stacks have different current sizes [0] and [1])
 [proguard] Error: Stacks have different current sizes [0] and [1]
```

Can be workarounded by using `-dontoptimize` Proguard option.

Proguard version 5.2.1
Kotlin version 1.1.2-4
